src/Transformation/Factory.php

```diff
@@ -0,0 +1,17 @@
+<?php
+/* Copyright (c) 2017 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Transformation;
+
+/**
+ * Factory for basic transformations.
+ */
+interface Factory {
+	/**
+	 * Create a custom transformation.
+	 *
+	 * @param	callable $f	mixed -> mixed
+	 * @return  Transformation
+	 */
+	public function custom(callable $f);
```

Comment:
To see how this factory will evolve and to get a better understanding of how this could reduce redundancy in transformations, it would be very useful to have at least one non-"custom" transformation.